src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -331,15 +339,7 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
 
         // Now stop each instance individually
         _ <- Future.traverse(cluster.nonPreemptibleInstances) { instance =>
-          // Install a startup script on the master node so Jupyter starts back up again once the instance is restarted
```

Comment:
Moved this installation from cluster stop time to cluster start time. I think it's better to set the cluster metadata lazily, right before it's started. This ensures that recent changes are applied to clusters that have been previously auto-paused.

Reply:
good call 👍

Reply:
Actually... I'm seeing more frequent test failures, "should pause and resume a cluster with preemptible instances" against this PR. I'm wondering if setting the instance metadata needs to be done at stop time, as it was before. I might move that back, but keep `setServiceAccount` at cluster start time.

Reply:
Hah...it's curious why that'd cause it